### PR TITLE
Cleanup error pages

### PR DESCRIPTION
**Pull request by Mugurell**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by Mugurell** [comment]
The biggest changes here are for the removal of the data url error page.

Everything should work, I tested on Fenix and RB for which I'll put up PRs tomorrow.

There are still some changes needed in `SystemEngineViewTest` which used `ErrorResponse.Content` (now deleted) but no `ErrorResponse.Uri` (the only response now remaining).
Will have to look why and how those tests could be refactored.

**Comment by Mugurell** [comment]
The tests from `SystemEngineViewTest` should now be fixed so this should be ready for review.

File: samples/browser/src/main/java/org/mozilla/samples/browser/request/SampleRequestInterceptor.kt [hunk 1]
```diff
@@ -1,56 +0,0 @@
...
-    ): InterceptionResponse? {
-        return when (uri) {
-            "sample:about" -> InterceptionResponse.Content("<h1>I am the sample browser</h1>")
-            else -> {
-                var response = context.components.appLinksInterceptor.onLoadRequest(
-                        engineSession, uri, lastUri, hasUserGesture, isSameDomain, isRedirect,
-                        isDirectNavigation, isSubframeRequest
-                )
-
-                if (response == null && !isDirectNavigation) {
```
Comment (csadilek) [inline samples/browser/src/main/java/org/mozilla/samples/browser/request/SampleRequestInterceptor.kt:36]: It seems like we'd be losing this PWA functionality in Sample Browser. Can we move it over to `SampleUrlEncodedRequestInterceptor` as well?

**Review by csadilek — APPROVED** [review]
Looks great! One comment below re: PWA new interceptor in Sample Browser.

File: samples/browser/src/main/java/org/mozilla/samples/browser/request/SampleRequestInterceptor.kt (hunk 1 above)
Comment (Mugurell) [reply samples/browser/src/main/java/org/mozilla/samples/browser/request/SampleRequestInterceptor.kt:36]: Thanks! Will look into it.

Comment (Mugurell) [reply samples/browser/src/main/java/org/mozilla/samples/browser/request/SampleRequestInterceptor.kt:36]: It was indeed a straightforward move of the code.

**Pull request merged**